lib/travis/api/build/app.rb

```diff
@@ -37,6 +27,35 @@ class App < Sinatra::Base
           if ENV.key?('LIBRATO_EMAIL') && ENV.key?('LIBRATO_TOKEN') && ENV.key?('LIBRATO_SOURCE')
             use Metriks
           end
+
+          use Rack::Deflater
+        end
+
+        helpers do
+          def auth_disabled?
+            (
+              ENV['API_TOKEN'].nil? || ENV['API_TOKEN'].strip.empty?
+            ) && (
+              settings.development? || settings.testing?
+            )
+          end
+        end
+
+        before '/script' do
+          return if auth_disabled?
+
+          unless env.key?('HTTP_AUTHORIZATION')
+            halt 401, 'missing Authorization header'
+          end
+
+          type, token = env['HTTP_AUTHORIZATION'].to_s.split(' ', 2)
+
+          ENV['API_TOKEN'].split(',').each do |valid_token|
+            return if Rack::Utils.secure_compare(type, 'token') &&
+                      Rack::Utils.secure_compare(token, valid_token)
```

Comment:
I'm confused about the utility of `Rack::Utils.secure_compare` if it does not do constant-time comparison 😐